packages/flame/lib/src/components/mixins/has_game_ref.dart

```diff
@@ -1,7 +1,7 @@
 import '../../../components.dart';
 import '../../../game.dart';
 
-mixin HasGameRef<T extends BaseGame> on Component {
+mixin HasGameRef<T extends Game> on Component {
```

Comment:
This can be simplified even more:
```dart
mixin HasGameRef<T> on Component {
  T? _gameRef;

  T get gameRef {
    if (_gameRef == null) {
      Component? c = parent;
      while (c != null) {
        if (c is HasGameRef<T> && c._gameRef != null) {
          _gameRef = c._gameRef;
          return _gameRef!;
        } else if (c is T) {
          _gameRef = c;
          return c;
        } else {
          c = c.parent;
        }
      }
      throw StateError("Cannot find reference $T in the component tree");
    }
    return _gameRef!;
  }

  @override
  void onRemove() {
    super.onRemove();
    _gameRef = null;
  }
}
```
Then neither `Game` nor `Component` will have to deal with the `HasGameRef`, it works on its own.
I suspect something like this can be done for `Tappable` and `Draggable` as well.

Reply:
Good idea!

Reply:
Glad to see that it actually worked! :)